scripts/packageComponent.js

```diff
@@ -10,62 +11,130 @@ const replace = require('replace-in-file')
+        })
+        .forEach(file => {
+          list.push(path.join('node_modules', 'ad-block', 'out', file))
+        })
+      break
+    case 'tracking-protection-updater':
+      list.push(path.join('node_modules', 'tracking-protection', 'data', 'TrackingProtection.dat'))
+      break
+    default:
+      throw new Error('Unrecognized component extension type: ' + componentType)
+  }
+
+  return list
+}
+
+function processDATFile (componentType, key, datFile) {
+  const datFileName = getNormalizedDATFileName(path.parse(datFile).name)
+  const stagingDir = path.join('build', componentType, datFileName)
+  const manifestsDir = getManifestsDirByComponentType(componentType)
+  const crxOutputDir = path.join('build', componentType)
+  const crxFile = path.join(crxOutputDir, `${componentType}-${datFileName}.crx`)
+
+  let privateKeyFile = ''
+
+  if (fs.lstatSync(key).isDirectory()) {
+    privateKeyFile = path.join(key, `${componentType}-${datFileName}.pem`)
+  } else {
+    privateKeyFile = key
+  }
+
+  stageFiles(datFile, manifestsDir, stagingDir)
+  generateCRXFile(crxFile, privateKeyFile, stagingDir, crxOutputDir)
+}
+
 installErrorHandlers()
 
 commander
-  .option('-k, --key <key>', 'private key file path', 'key.pem')
+  .option('-d, --keys-directory <dir>', 'directory containing private keys for signing crx files')
+  .option('-f, --key-file <file>', 'private key file for signing crx', 'key.pem')
   .option('-s, --set-version <x.x.x>', 'component extension version number')
   .option('-t, --type <type>', 'component extension type', /^(ad-block-updater|https-everywhere-updater|tracking-protection-updater)$/i, 'ad-block-updater')
   .parse(process.argv)
 
-if (!fs.existsSync(commander.key)) {
-  throw new Error('Missing private key file: ' + commander.key)
-}
+let keyParam = ''
 
-if (!commander.setVersion) {
-  throw new Error('Missing required option: --set-version')
+if (fs.existsSync(commander.keyFile)) {
+  keyParam = commander.keyFile
+} else if (fs.existsSync(commander.keysDirectory)) {
+  keyParam = commander.keysDirectory
+} else {
+  throw new Error('Missing or invalid private key file/directory')
 }
 
-if (!commander.setVersion.match(/^(\d+\.\d+\.\d+)$/)) {
+if (!commander.setVersion || !commander.setVersion.match(/^(\d+\.\d+\.\d+)$/)) {
   throw new Error('Missing or invalid option: --set-version')
 }
 
-const outputDir = path.join('build', commander.type)
-const inputDir = path.join('build', commander.type)
-const datFile = getDATFileFromComponentType(commander.type)
-const crxFile = path.join(outputDir, commander.type + '.crx')
-const privateKeyFile = commander.key
+generateManifestFilesByComponentType(commander.type)
 
-stageFiles(commander, outputDir, datFile)
-generateCRXFile(crxFile, privateKeyFile, inputDir, outputDir)
+getDATFileListByComponentType(commander.type).forEach(datFile => {
+  processDATFile(commander.type, keyParam, datFile)
+})
```

Comment:
I think you could do
`getDATFileListByComponentType(commander.type).forEach(processDATFile.bind(null, commander.type, keyParam))`